http://bugs.gentoo.org/130827

2006-03-31  Richard Guenther  <rguenther@suse.de>

	PR bootstrap/26936
	Backport from mainline.
	2006-02-25  Juergen Weigert  <jw@suse.de>
		Richard Guenther  <rguenther@suse.de>
 
	* scan-decls.c (scan_decls): Don't fetch new statement after CPP_EOF.

Index: gcc-4_1-branch/gcc/scan-decls.c
===================================================================
--- gcc-4_1-branch/gcc/scan-decls.c	(revision 112572)
+++ gcc-4_1-branch/gcc/scan-decls.c	(revision 112573)
@@ -202,6 +202,8 @@ scan_decls (cpp_reader *pfile, int argc 
 		 parameter list */
 	      while (token->type != CPP_SEMICOLON && token->type != CPP_EOF)
 		token = get_a_token (pfile);
+	      if (token->type == CPP_EOF)
+		return 0;
 	      goto new_statement;
 	    }
 	  break;
